msgraph-mail/php/src/Users/Item/InferenceClassification/Overrides/OverridesRequestBuilder.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace Microsoft\Graph\Users\Item\InferenceClassification\Overrides;
+
+use \Exception;
+use Http\Promise\Promise;
+use Http\Promise\RejectedPromise;
+use Microsoft\Graph\Models\Microsoft\Graph\InferenceClassificationOverride;
+use Microsoft\Kiota\Abstractions\HttpMethod;
+use Microsoft\Kiota\Abstractions\MiddlewareOption;
+use Microsoft\Kiota\Abstractions\RequestAdapter;
+use Microsoft\Kiota\Abstractions\RequestInformation;
+use Microsoft\Kiota\Abstractions\ResponseHandler;
+use Microsoft\Kiota\Abstractions\Serialization\Parsable;
+
+class OverridesRequestBuilder 
+{
+    /** @var array|null $pathParameters Path parameters for the request */
+    private array $pathParameters;
+    
+    /** @var RequestAdapter|null $requestAdapter The request adapter to use to execute the requests. */
+    private RequestAdapter $requestAdapter;
+    
+    /** @var string|null $urlTemplate Url template to use to build the URL for the current request builder */
+    private ?string $urlTemplate;
+    
+    /**
+     * Instantiates a new OverridesRequestBuilder and sets the default values.
+     * @param array $pathParameters Path parameters for the request
+     * @param RequestAdapter $requestAdapter The request adapter to use to execute the requests.
+    */
+    public function __construct(array $pathParameters, RequestAdapter $requestAdapter) {
+        $this->urlTemplate = '{+baseurl}/users/{user_id}/inferenceClassification/overrides{?top,skip,search,filter,count,orderby,select,expand}';
+        $this->requestAdapter = $requestAdapter;
+        $this->pathParameters = $pathParameters;
```

Comment:
we should make a copy of the array (assuming the array is passed by reference and not by value copy here), otherwise we'll end up having parameters set across multiple requests. (they all come from the initial array in the api client)

Reply:
Parameters are passed by value unless you prefix it with a `&`. In this case `array &$pathParameters` passes the parameter by reference.

Reply:
thanks for the precision!